datalad/distribution/add_sibling.py

```diff
@@ -0,0 +1,183 @@
+from datalad.support.param import Parameter
+from datalad.support.constraints import EnsureStr, EnsureNone
+from datalad.support.gitrepo import GitRepo
+from datalad.cmd import Runner
+from ..interface.base import Interface
+from datalad.distribution.dataset import EnsureDataset, Dataset, datasetmethod
+from datalad.utils import getpwd
+
+
+lgr = logging.getLogger('datalad.distribution.add_publication_target')
+
+
+class AddSibling(Interface):
+    """Adds a sibling to a dataset."""
+
+    _params_ = dict(
+        # TODO: Somehow the replacement of '_' and '-' is buggy on
+        # positional arguments
+        dataset=Parameter(
+            args=("--dataset", "-d",),
+            doc="""specify the dataset to add the sibling to. If
+                no dataset is given, an attempt is made to identify the dataset
+                based on the current working directory""",
+            constraints=EnsureDataset() | EnsureNone()),
+        sibling=Parameter(
+            args=('sibling',),
+            doc="""Name of the sibling to be added. If RECURSIVE is set, the
+                same name will be used to address the subdatasets' siblings""",
+            constraints=EnsureStr() | EnsureNone()),
+        url=Parameter(
+            args=('url',),
+            doc="""The URL of or path to the dataset sibling named by
+                `sibling`.
+                If you want to recursively add siblings, it is expected, that
+                you pass a template for building the URLs of the siblings of
+                all (sub)datasets by using placeholders.\n
+                List of currently available placeholders:\n
+                %%NAME\tthe name of the dataset, where slashes are replaced by
+                dashes.\nThis option is ignored if there is already a
+                configured sibling dataset under the name given by `sibling`""",
+            constraints=EnsureStr() | EnsureNone()),
+        pushurl=Parameter(
+            args=('--pushurl',),
+            doc="""In case the `url` cannot be used to publish to the dataset
+                sibling, this option specifies a URL to be used instead.\n
+                This option is ignored if there is already a configured sibling
+                dataset under the name given by `sibling`""",
+            constraints=EnsureStr() | EnsureNone()),
+        recursive=Parameter(
+            args=("--recursive", "-r"),
+            action="store_true",
+            doc="""Recursively add the `sibling` to all subdatasets of
+                `dataset`""",),
+        force=Parameter(
+            args=("--force", "-f",),
+            action="store_true",
+            doc="""If `sibling` exists already, force to (re-)configure its
+                URLs""",),)
+
+    @staticmethod
+    @datasetmethod(name='add_sibling')
+    def __call__(dataset=None, sibling=None, url=None,
+                 pushurl=None, recursive=False, force=False):
+
+        # TODO: Detect malformed URL and fail?
```

Comment:
I think, it should be worth it later on. The point is: We planned on having a "sibling validation" for the dataset class. It would be quite confusing, if we are able to validate on getting siblings, but not at all on adding them. So, at some point we should provide that option here, too.